src/main/scala/net/psforever/actors/session/support/SessionAvatarHandlers.scala

```diff
@@ -0,0 +1,545 @@
+                             chatActor: typed.ActorRef[ChatActor.Command],
+                             implicit val context: ActorContext
+                           ) extends CommonSessionInterfacingFunctionality {
+  /**
+   * na
+   *
+   * @param toChannel na
+   * @param guid      na
+   * @param reply     na
+   */
+  def handle(toChannel: String, guid: PlanetSideGUID, reply: AvatarResponse.Response): Unit = {
+    val tplayer_guid =
+      if (player != null && player.HasGUID) player.GUID
+      else PlanetSideGUID(0)
+    reply match {
+      case AvatarResponse.TeardownConnection() =>
+        log.trace(s"ending ${player.Name}'s old session by event system request (relog)")
+        context.stop(context.self)
+
+      case AvatarResponse.SendResponse(msg) =>
+        sendResponse(msg)
+
+      case AvatarResponse.SendResponseTargeted(target_guid, msg) =>
+        if (tplayer_guid == target_guid) {
+          sendResponse(msg)
+        }
+
+      case AvatarResponse.Revive(target_guid) =>
+        if (tplayer_guid == target_guid) {
+          log.info(s"No time for rest, ${player.Name}.  Back on your feet!")
+          sessionData.zoning.spawn.reviveTimer.cancel()
+          sessionData.zoning.spawn.deadState = DeadState.Alive
+          player.Revive
+          val health = player.Health
+          sendResponse(PlanetsideAttributeMessage(target_guid, 0, health))
+          sendResponse(AvatarDeadStateMessage(DeadState.Alive, 0, 0, player.Position, player.Faction, unk5=true))
+          continent.AvatarEvents ! AvatarServiceMessage(
+            continent.id,
+            AvatarAction.PlanetsideAttributeToAll(target_guid, 0, health)
+          )
+        }
+
+      case AvatarResponse.ArmorChanged(suit, subtype) =>
+        if (tplayer_guid != guid) {
+          sendResponse(ArmorChangedMessage(guid, suit, subtype))
+        }
+
+      case AvatarResponse.ChangeAmmo(weapon_guid, weapon_slot, previous_guid, ammo_id, ammo_guid, ammo_data) =>
+        if (tplayer_guid != guid) {
+          sendResponse(ObjectDetachMessage(weapon_guid, previous_guid, Vector3.Zero, 0))
+          sendResponse(
+            ObjectCreateMessage(
+              ammo_id,
+              ammo_guid,
+              ObjectCreateMessageParent(weapon_guid, weapon_slot),
+              ammo_data
+            )
+          )
+          sendResponse(ChangeAmmoMessage(weapon_guid, 1))
+        }
+
+      case AvatarResponse.ChangeFireMode(item_guid, mode) =>
+        if (tplayer_guid != guid) {
+          sendResponse(ChangeFireModeMessage(item_guid, mode))
+        }
+
+      case AvatarResponse.ChangeFireState_Start(weapon_guid) =>
+        if (tplayer_guid != guid) {
+          sendResponse(ChangeFireStateMessage_Start(weapon_guid))
+        }
+
+      case AvatarResponse.ChangeFireState_Stop(weapon_guid) =>
+        if (tplayer_guid != guid) {
+          sendResponse(ChangeFireStateMessage_Stop(weapon_guid))
+        }
```

Comment:
compact all of those :

```scala 
case a if something(a) => doThis()
```

instead of:

```scala
case a => 
if (something(a) {
doThis()
}
```